slam-libraries/viam-orb-slam3/orbslam_server_v1_test.cc

```diff
@@ -100,26 +126,37 @@ BOOST_AUTO_TEST_CASE(ParseAndValidateArguments_valid_config) {
     BOOST_TEST(slamService.map_rate_sec.count() == chrono::seconds(60).count());
     BOOST_TEST(slamService.camera_name == "color");
     BOOST_TEST(slamService.offlineFlag == false);
+    BOOST_TEST(slamService.delete_processed_data == false);
 }
 
 BOOST_AUTO_TEST_CASE(
     ParseAndValidateArguments_valid_config_capitalized_slam_mode) {
-    const vector<string> args{"-data_dir=/path/to", "-config_param={mode=RGBD}",
-                              "-port=20000",        "-sensors=color",
-                              "-data_rate_ms=200",  "-map_rate_sec=60"};
+    const vector<string> args{"-data_dir=/path/to",
+                              "-config_param={mode=RGBD}",
+                              "-port=20000",
+                              "-sensors=color",
+                              "-data_rate_ms=200",
+                              "-map_rate_sec=60",
+                              "-delete_processed_data=true"};
     SLAMServiceImpl slamService;
     utils::ParseAndValidateArguments(args, slamService);
     BOOST_TEST(slamService.slam_mode == "rgbd");
+    BOOST_TEST(slamService.delete_processed_data == true);
 }
 
 BOOST_AUTO_TEST_CASE(ParseAndValidateArguments_valid_config_no_camera) {
-    const vector<string> args{
-        "-data_dir=/path/to", "-config_param={mode=rgbd}", "-port=20000",
-        "-sensors=",          "-data_rate_ms=200",         "-map_rate_sec=60"};
+    const vector<string> args{"-data_dir=/path/to",
+                              "-config_param={mode=rgbd}",
+                              "-port=20000",
+                              "-sensors=",
+                              "-data_rate_ms=200",
+                              "-map_rate_sec=60",
+                              "-delete_processed_data=false"};
     SLAMServiceImpl slamService;
     utils::ParseAndValidateArguments(args, slamService);
     BOOST_TEST(slamService.camera_name == "");
     BOOST_TEST(slamService.offlineFlag == true);
+    BOOST_TEST(slamService.delete_processed_data == false);
 }
```

Comment:
[minor] Instead of incorporating the `delete_processed_data` tests indirectly in other tests, break these out into a separate test case where you check for all four combinations of situations: offline/online vs. delete_processed_data=false/true.